addons/binding/org.openhab.binding.solarlog/src/main/java/org/openhab/binding/solarlog/handler/SolarLogHandler.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    private void refresh() throws Exception {
+        // Get the JSON - somehow
+        JsonElement solarLogDataElement = HttpUtils.getSolarLogData(config.url);
+        JsonObject solarLogData = solarLogDataElement.getAsJsonObject();
+
+        // Check whether the data is well-formed
+        if (solarLogData.has(SolarLogBindingConstants.SOLARLOG_JSON_ROOT)) {
+            solarLogData = solarLogData.getAsJsonObject(SolarLogBindingConstants.SOLARLOG_JSON_ROOT);
+            if (solarLogData.has(SolarLogBindingConstants.SOLARLOG_JSON_PROPERTIES)) {
+                solarLogData = solarLogData.getAsJsonObject(SolarLogBindingConstants.SOLARLOG_JSON_PROPERTIES);
+
+                for (SolarLogChannel channelConfig : SolarLogChannel.values()) {
+                    if (solarLogData.has(channelConfig.getIndex())) {
+                        String value = solarLogData.get(channelConfig.getIndex()).getAsString();
+                        Channel channel = getThing().getChannel(channelConfig.getId());
+                        State state = getState(value, channelConfig);
+
+                        updateState(channel.getUID(), state);
+                    } else {
+                        logger.debug("Error refreshing source {}", getThing().getUID(), channelConfig.getId());
+                    }
+                }
+            }
+        }
+
+    }
+
+    private State getState(String value, SolarLogChannel type) {
+        switch (type) {
+            // Only DateTime channel
+            case CHANNEL_LASTUPDATETIME:
+                try {
+                    logger.trace("Parsing date {}", value);
+                    try {
+                        Date date = new SimpleDateFormat("dd.MM.yy HH:mm:ss").parse(value);
+                        SimpleDateFormat sdfDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");// dd/MM/yyyy
+                        String strDate = sdfDate.format(date);
+
+                        logger.trace("Parsing date successful. Returning date. {}", new DateTimeType(strDate));
+                        return new DateTimeType(strDate);
+                    } catch (ParseException fpe) {
+                        logger.trace("Parsing date failed. Returning string.", fpe);
+                        return new StringType(value);
+                    }
+                } catch (IllegalArgumentException e) {
+                    logger.trace("Parsing date failed. Returning string", e);
+                    return new StringType(value);
+                }
+                // All other channels should be numbers
+            default:
+                try {
+                    logger.trace("Parsing number {}", value);
+                    return new DecimalType(new BigDecimal(value));
+                } catch (NumberFormatException e) {
+                    logger.trace("Parsing number failed. Returning string");
+                    return new StringType(value);
```

Comment:
oh, hell - this looks dangerous. All your channels expect DecimalTypes, so passing them a StringType all of a sudden isn't a sensible thing to do. Rather don't send any channel update in that case (and maybe even log it as a warning or error, if you consider this situation a bug in your binding that should be reported).